addons/l10n_it_edi/models/res_partner.py

```diff
@@ -135,8 +135,10 @@ def _l10n_it_edi_normalized_codice_fiscale(self, l10n_it_codice_fiscale=None):
         if l10n_it_codice_fiscale is None:
             self.ensure_one()
             l10n_it_codice_fiscale = self.l10n_it_codice_fiscale
-        if l10n_it_codice_fiscale and re.match(r'^IT[0-9]{11}$', l10n_it_codice_fiscale):
-            return l10n_it_codice_fiscale[2:13]
+        if l10n_it_codice_fiscale:
+            l10n_it_codice_fiscale = l10n_it_codice_fiscale.upper()
+            if re.match(r'^IT[0-9]{11}$', l10n_it_codice_fiscale):
+                return l10n_it_codice_fiscale[2:13]
         return l10n_it_codice_fiscale
```

Comment:
```suggestion
    def _l10n_it_edi_normalized_codice_fiscale(self, l10n_it_codice_fiscale=None):
        from stdnum.it.iva import validate
        try:
            return validate(l10n_it_codice_fiscale or self.l10n_it_codice_fiscale)
        except Exception as e:
            raise UserError(_("Invalid VAT")) from e
```
I think this entire method could be made much simpler (and more robust) by using the dedicated `stdnum` module :nerd_face:

Reply:
The "Codice Fiscale" is different from the VAT number for individuals.
For companies, it is indeed the 11 numbers of the VAT number, but for individuals it's 16 alphanumeric characters ("MRTMTT91D08F205J" for example).
`stdnum.it.iva` is already used in `base_vat` to check if the VAT is valid.
`stdnum.it.codicefiscale` does exist and is used in Odoo to validate the partner form.

In general, I don't think it's the goal of this method to validate the number, its purpose is only to normalize the code so that it gets accepted by the SDI.

Reply:
Then maybe simply using `compact()` should be enough?

Reply:
Indeed, I did not notice this function.
It can only replace the line with `upper()` as it will not remove the 'IT' but it will be more robust that way.
I will update the commit.

Reply:
If you use `compact()` from `stdnum.it.iva` it should do both, no? :thinking:

Reply:
Indeed, but also for individuals, which is not wanted. 
The three first letters are based on the surname of the individual, so I think they can start with "IT".

Reply:
Right, I keep forgetting about individuals when thinking about VAT :sweat_smile: